### PR TITLE
Refs [TASK][34061] Handle event Add/Edit Note Screen

### PR DESCRIPTION
**Pull request by trangnq258**
## Related Tickets
- [#Task 34061: Handle event Add/Edit Note Screen](https://edu-redmine.sun-asterisk.vn/issues/34061)
## WHAT
- Add Note
 +) Button agree click: check information -> add note 
 +) Button cancel: close screen
- Edit Note: show title, description of selected note
 +) Button agree click: check infomation ->update note
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/yummy/ui/note/AddNoteFragment.kt [hunk 1]
```diff
@@ -2,19 +2,59 @@ package com.example.yummy.ui.note
...
     override fun initActions() {
+        imageButtonCancel.setOnClickListener {
+            closeFragment()
+        }
+        imageButtonOk.setOnClickListener {
+            if (editTextName.text.isEmpty()) {
+                editTextName.error = getString(R.string.error_not_empty)
+            } else {
+                notePresenter?.insertNote(
+                    Note(0, editTextName.text.toString(), editTextDescription.text.toString(), "")
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/note/AddNoteFragment.kt]: thế cái id cứ để là 0 thì có tác dụng gì nhỉ ?

File: app/src/main/java/com/example/yummy/ui/note/EditNoteFragment.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                editTextName.error = getString(R.string.error_not_empty)
+            } else if (editTextName.text.toString() != note?.title || editTextDescription.text.toString() != note?.description) {
+                note?.apply {
+                    title = editTextName.text.toString()
+                    description = editTextDescription.text.toString()
+                    notePresenter?.updateNote(this)
+                }
+            } else {
+                closeFragment()
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/note/EditNoteFragment.kt]: 3 case trở lên thì dùng when 
hoặc dùng if(){... return }

File: app/src/main/java/com/example/yummy/ui/note/AddNoteFragment.kt (hunk 1 above)
Comment (trangnq258) [reply app/src/main/java/com/example/yummy/ui/note/AddNoteFragment.kt]: khi add note thì id của note sẽ tự động tăng ạ

File: app/src/main/java/com/example/yummy/data/model/Note.kt [hunk 3]
```diff
@@ -9,10 +9,10 @@ import java.util.*
...
 @Parcelize
 data class Note(
-    val id: Int,
-    var title: String,
-    var description: String,
-    private val date: String
+    val id: Int = DEFAULT_ID,
+    var title: String = DEFAULT_TITLE,
+    var description: String = DEFAULT_DESCRIPTION,
+    private val date: String = DEFAULT_DATE
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/data/model/Note.kt]:
```suggestion
    val id: Int = 0,
    var title: String ="",
    var description: String ="",
    private val date: String =""
```

**Pull request merged**